src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -891,9 +891,15 @@ class FrontendGenImpl {
     if (nOps == 2) {
       llvm::SmallVector<int64_t, 2> dims;
       dims.push_back(1);
-      llvm::SmallVector<float, 2> values;
-      values.push_back(0.);
-      auto elementType = builder_.getF32Type();
+
+      std::vector<Value> inputs;
+      getNodeInputs(node, inputs);
+      auto elementType =
+          inputs[0].getType().cast<TensorType>().getElementType();
+
+      llvm::SmallVector<Attribute, 2> values(
+          1, builder_.getZeroAttr(elementType));
+
```

Comment:
The original code tried to use tensor<0xT> to represent none. Do you want to use a different way to create the none?

Reply:
I haven't changed the underlying tensor type aside from the element type, in the original code it would be tensor<1xf32> (with value of 0.0), after the change it becomes tensor<1xT> (with value of (T)(0.0)).

I think there could maybe be an argument for: disallowing the NoneType all together for the input (since effectively it's being handle here), or allowing it and maybe breaking out the handling from this portion, thoughts?

Reply:
You are right that this optional input has a default value.